pyro/infer/enum.py

```diff
@@ -30,16 +40,67 @@ def iter_discrete_traces(graph_type, fn, *args, **kwargs):
     """
     queue = LifoQueue()
     queue.put(Trace())
-    q_fn = poutine.queue(fn, queue=queue)
+    q_fn = poutine.queue(fn, queue=queue, escape_fn=_iter_discrete_escape)
     while not queue.empty():
-        q_fn = poutine.queue(fn, queue=queue)
-        full_trace = poutine.trace(
-            q_fn, graph_type=graph_type).get_trace(*args, **kwargs)
+        full_trace = poutine.trace(q_fn, graph_type=graph_type).get_trace(*args, **kwargs)
 
         # Scale trace by probability of discrete choices.
-        log_pdf = full_trace.batch_log_pdf(site_filter=site_is_discrete)
+        log_pdf = 0
+        full_trace.compute_batch_log_pdf(site_filter=_iter_discrete_filter)
+        for name, site in full_trace.nodes.items():
+            if _iter_discrete_filter(name, site):
+                log_pdf = log_pdf + sum_rightmost(site["batch_log_pdf"], max_iarange_nesting)
         if isinstance(log_pdf, Variable):
             scale = torch.exp(log_pdf.detach())
         else:
             scale = math.exp(log_pdf)
+
         yield scale, full_trace
+
+
+def _config_enumerate(default):
+
+    def config_fn(site):
+        if site["type"] != "sample" or site["is_observed"]:
+            return {}
+        if not getattr(site["fn"], "enumerable", False):
+            return {}
+        if "enumerate" in site["infer"]:
+            return {}  # do not overwrite existing config
+        return {"enumerate": default}
+
+    return config_fn
+
+
+def config_enumerate(guide=None, default="sequential"):
+    """
+    Configures each enumerable site a guide to enumerate with given method,
```

Comment:
should the docstring mention that it doesn't override?

Reply:
Sure